lib/path.js

```diff
@@ -1533,6 +1533,19 @@ const posix = {
   posix: null
 };
 
+let urlModule;
+function fromURL(fileURLOrPath) {
+  // Lazy require to avoid circular dependency
+  urlModule ??= require('internal/url');
+  if (!urlModule.isURLInstance(fileURLOrPath)) {
+    validateString(fileURLOrPath, 'fileURLOrPath');
+  }
+
+  return urlModule.toPathIfFileURL(fileURLOrPath);
+}
+
+posix.fromURL = win32.fromURL = fromURL;
```

Comment:
This seems wrong. `posix` and `win32` should provide implementations that do not depend on the actual OS, but the behavior of `fromURL()` seems to depend on the OS.

Reply:
`node:url` does not implement separate `posix` and `win32` APIs, but `node:path` does. `path.win32.fromURL()` should behave as if the underlying OS was Windows even when the actual OS is posix.